Mathlib/Algebra/Polynomial/Bivariate.lean

```diff
@@ -41,15 +41,101 @@ abbrev CC (r : R) : R[X][Y] := C (C r)
 lemma evalEval_C (x y : R) (p : R[X]) : (C p).evalEval x y = p.eval x := by
   rw [evalEval, eval_C]
 
+@[simp]
+lemma evalEval_CC (x y : R) (p : R) : (CC p).evalEval x y = p := by
+  rw [evalEval_C, eval_C]
+
+@[simp]
+lemma evalEval_zero (x y : R) : (0 : R[X][Y]).evalEval x y = 0 := by
+  simp only [evalEval, eval_zero]
+
+@[simp]
+lemma evalEval_one (x y : R) : (1 : R[X][Y]).evalEval x y = 1 := by
+  simp only [evalEval, eval_one]
+
+@[simp]
+lemma evalEval_ofNat (x y : R) (n : ℕ) [n.AtLeastTwo] :
+    (OfNat.ofNat n : R[X][Y]).evalEval x y = OfNat.ofNat n := by
```

Comment:
Is it necessary to write OfNat.ofNat here, rather than just n?

Reply:
I copied the corresponding lemma for `eval`. Without the `Ofnat.ofNat` this is the same as `evalEval_natCast`.